Error: The server does not support SSL connections

### Describe the bug

Unable to deploy Unleash with built-in Postgres DB with default DB values.

Helm Chart: unleash
Helm Chart Version: 4.1.1

### Steps to reproduce the bug

1. `helm repo add unleash https://docs.getunleash.io/helm-charts`
2. `helm template unleash unleash/unleash`
3. `helm install unleash unleash/unleash`
4. check Unleash pod logs

### Expected behavior

Unleash migrate a postgres DB and started successfully.

### Logs, error output, etc.

```bash
[2024-05-10T16:01:04.890] [ERROR] server-impl.js - Locking error: The server does not support SSL connections
[2024-05-10T16:01:04.892] [ERROR] server-impl.js - Failed to migrate db Error: The server does not support SSL connections
    at Socket.<anonymous> (/unleash/node_modules/pg/lib/connection.js:77:37)
    at Object.onceWrapper (node:events:632:26)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:308:9)
    at Readable.push (node:internal/streams/readable:245:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: The server does not support SSL connections
    at Socket.<anonymous> (/unleash/node_modules/pg/lib/connection.js:77:37)
    at Object.onceWrapper (node:events:632:26)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:308:9)
    at Readable.push (node:internal/streams/readable:245:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)

Node.js v18.18.2
```


### Screenshots

_No response_

### Additional context

Most likely the issue is related to this change https://github.com/Unleash/helm-charts/pull/139/files#diff-9576a5f35de34ba1784129471c2262fc5bc8bd10b69c258b07f25e1a0d1f5d44R51-R54
When I downgraded to Helm Chart Version 4.0.2, Unleash started without any errors

### Unleash version

_No response_

### Subscription type

Open source

### Hosting type

Self-hosted

### SDK information (language and version)

_No response_

--- a/charts/unleash/Chart.yaml
+++ b/charts/unleash/Chart.yaml
@@ -5,7 +5,7 @@ icon: https://docs.getunleash.io/img/logo.svg
 
 type: application
 
-version: 5.1.4
+version: 5.1.5
 
 appVersion: "6.0.6"
 

--- a/charts/unleash/templates/deployment.yaml
+++ b/charts/unleash/templates/deployment.yaml
@@ -52,7 +52,7 @@ spec:
             {{- if .Values.dbConfig.ssl }}
             - name: DATABASE_SSL
               value: {{ .Values.dbConfig.ssl | toJson | quote }}
-            {{- else if .Values.postgresql.enabled }}
+            {{- else }}
             - name: DATABASE_SSL
               value: "false"
             {{- end }}

--- a/charts/unleash/templates/deployment.yaml
+++ b/charts/unleash/templates/deployment.yaml
@@ -52,7 +52,7 @@ spec:
             {{- if .Values.dbConfig.ssl }}
             - name: DATABASE_SSL
               value: {{ .Values.dbConfig.ssl | toJson | quote }}
-            {{- else if (not .Values.dbConfig.sslConfigFile) and (not .Values.dbConfig.sslCaFile) and (not .Values.dbConfig.sslKeyFile) and (not .Values.dbConfig.sslCertFile) and (not .Values.dbConfig.sslRejectUnauthorized) }}
+            {{- else if and (not .Values.dbConfig.sslConfigFile) (not .Values.dbConfig.sslCaFile) (not .Values.dbConfig.sslKeyFile) (not .Values.dbConfig.sslCertFile) (not .Values.dbConfig.sslRejectUnauthorized) }}
             - name: DATABASE_SSL
               value: "false"
             {{- end }}

--- a/charts/unleash/templates/deployment.yaml
+++ b/charts/unleash/templates/deployment.yaml
@@ -52,7 +52,7 @@ spec:
             {{- if .Values.dbConfig.ssl }}
             - name: DATABASE_SSL
               value: {{ .Values.dbConfig.ssl | toJson | quote }}
-            {{- else }}
+            {{- else if (not .Values.dbConfig.sslConfigFile) and (not .Values.dbConfig.sslCaFile) and (not .Values.dbConfig.sslKeyFile) and (not .Values.dbConfig.sslCertFile) and (not .Values.dbConfig.sslRejectUnauthorized) }}
             - name: DATABASE_SSL
               value: "false"
             {{- end }}

